### PR TITLE
Rework the project list in home tab

### LINKED ISSUE #3214: After sucessfull synchronization, home page is not refreshed
After the successful synchronization of an updated project through home screen buttons, there is 

1. no notification about the successful finish of the synchronization
2. there is still "Pending changes to synchronize" message on the project
3. You can still see the "Synchronize button" under 3 dots  

https://github.com/MerginMaps/mobile/assets/83582181/2b297d7f-c69e-421f-95d5-832d1436f17c

### PR DESCRIPTION
**Pull request by uclaros**
Fixes #3214

Change the way the active project is displayed on top of the list in Home tab into a less hacky one.

Now the ProjectsModel gets a `activeProjectId` property which _if_ set, then one can use the new `Roles::ProjectIsActiveProject` role on the model's contents.

We then split the projects list into two sections based on this new role.

`Roles::ProjectIsActiveProject` may be a bit misleading, as it might not return true on the active project, rather on the one passed into `ProjectsModel::activeProjectId`

### CODE REVIEW

File: app/qml/project/components/MMProjectDelegate.qml [hunk 1]
```diff
@@ -40,8 +40,11 @@ Control {
 
   topPadding: __style.margin20
   rightPadding: __style.margin20
-  bottomPadding: __style.margin20
   leftPadding: __style.margin20
+  // The last item in a section ('currently open' vs 'downloaded projects' in Home tab) gets an extra
+  // 40 * __dp padding, then the inset is also grown to keep the background at the correct size
+  bottomPadding: ListView.section !== ListView.nextSection ? __style.margin20 + 40 * __dp : __style.margin20
```
Comment (tomasMizera) [inline app/qml/project/components/MMProjectDelegate.qml:46]:
```suggestion
  bottomPadding: ListView.section !== ListView.nextSection ? __style.margin40 : __style.margin20
```

File: app/qml/project/components/MMProjectDelegate.qml [hunk 2]
```diff
@@ -40,8 +40,11 @@ Control {
...
+  bottomInset: ListView.section !== ListView.nextSection ? 40 * __dp : topInset
```
Comment (tomasMizera) [inline app/qml/project/components/MMProjectDelegate.qml:47]:
```suggestion
  bottomInset: ListView.section !== ListView.nextSection ? __style.margin40 : topInset
```

File: app/qml/project/MMProjectHomeTab.qml [hunk 3]
```diff
@@ -230,29 +116,39 @@ Item {
...
+    activeProjectAlwaysFirst: true
+    projectsModel.activeProjectId: root.activeProjectId
+
+    list.section {
+      property: "ProjectIsActiveProject"
+      criteria: ViewSection.FullString
+      delegate: Column {
+        width: ListView.view.width
+
+        Text {
```
Comment (tomasMizera) [inline app/qml/project/MMProjectHomeTab.qml:128]: Can we use `MMText` here, please? It automatically sets the line height (you should not need to set `height` explicitly then)

File: app/qml/project/MMProjectHomeTab.qml [hunk 4]
```diff
@@ -230,29 +116,39 @@ Item {
...
 
+    activeProjectAlwaysFirst: true
+    projectsModel.activeProjectId: root.activeProjectId
+
+    list.section {
+      property: "ProjectIsActiveProject"
+      criteria: ViewSection.FullString
+      delegate: Column {
+        width: ListView.view.width
+
```
Comment (tomasMizera) [inline app/qml/project/MMProjectHomeTab.qml:127]: Let's not use the default value of the `spacing` property of `Column`. Please set it explicitly.

File: app/projectsproxymodel.h [hunk 5]
```diff
@@ -52,6 +57,7 @@ class ProjectsProxyModel : public QSortFilterProxyModel
     ProjectsModel *mModel = nullptr; // not owned by this, needs to be set in order to proxy model to work
     ProjectsModel::ProjectModelTypes mModelType = ProjectsModel::EmptyProjectsModel;
     QString mSearchExpression;
+    bool mactiveProjectAlwaysFirst = false;
```
Comment (tomasMizera) [inline app/projectsproxymodel.h:60]: just a code convention
```suggestion
    bool mActiveProjectAlwaysFirst = false;
```

**Review by tomasMizera — COMMENTED** [review]
This is looking great! 🚀

**Review by PeterPetrik — APPROVED** [review]

**Pull request merged**